src/mds/MDCache.cc

```diff
@@ -829,6 +829,20 @@ MDSCacheObject *MDCache::get_object(const MDSCacheObjectInfo &info)
 }
 
 
+// ====================================================================
+// consistent hash ring
+
+mds_rank_t hash_into_rank_bucket(inodeno_t ino, mds_rank_t max_mds)
+{
+  uint64_t hash = rjhash64(ino);
+  int64_t b = -1, j = 0;
+  while (j < max_mds) {
+    b = j;
+    hash = hash*2862933555777941757ULL + 1;
+    j = (b + 1) * (double(1LL << 31) / double((key >> 33) + 1));
+  }
```

Comment:
Please explain this method more. It needs comments for the magic numbers especially.

Reply:
@batrick  I believe this comes directly from Lamping & Veach.  See the implementation of JumpConsistentHash:

https://arxiv.org/pdf/1406.2294.pdf

Reply:
Sure, I suspected as much but it must the source must be cited and attribution given!